sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobAsyncClient.java

```diff
@@ -423,205 +430,151 @@ public void onFailure(Throwable throwable, Response response) {
-     * @param metadata          Metadata to associate with the blob.
-     * @param cpkInfo           Additional parameters for the operation.
-     * @param cancellationToken The token to request cancellation.
-     * @param callback          Callback that receives the response.
+     * @param options {@link BlobSetMetadataOptions}
+     * @param callback Callback that receives the response.
      */
-    public void setBlobMetadata(String containerName,
-                                String blobName,
-                                Integer timeout,
-                                BlobRequestConditions requestConditions,
-                                Map<String, String> metadata,
-                                CpkInfo cpkInfo,
-                                CancellationToken cancellationToken,
-                                CallbackWithHeader<Void, BlobSetMetadataHeaders> callback) {
-        storageBlobServiceClient.setBlobMetadata(containerName,
-            blobName,
-            timeout,
-            requestConditions,
-            metadata,
-            cpkInfo,
-            cancellationToken,
-            callback);
+    public void setBlobMetadata(@NonNull BlobSetMetadataOptions options,
+                                @Nullable CallbackWithHeader<Void, BlobSetMetadataHeaders> callback) {
+        Objects.requireNonNull(options);
+        storageBlobServiceClient.setBlobMetadata(options.getContainerName(), options.getBlobName(),
+            options.getTimeout(), options.getRequestConditions(), options.getMetadata(), options.getCpkInfo(),
+            options.getCancellationToken(), callback);
     }
 
     /**
-     * Sets the blob's tier.
+     * Sets the tier on a blob. The operation is allowed on a page blob in a premium storage account or a block blob in
+     * a blob storage or GPV2 account. A premium page blob's tier determines the allowed size, IOPS, and bandwidth of
+     * the blob. A block blob's tier determines the Hot/Cool/Archive storage type. This does not update the blob's
+     * etag.
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/set-blob-tier">Azure Docs</a></p>
      *
      * @param containerName The container name.
      * @param blobName      The blob name.
      * @param tier          The access tier.
      * @param callback      The callback that receives the response.
      */
-    public void setBlobTier(String containerName,
-                            String blobName,
-                            AccessTier tier,
-                            CallbackWithHeader<Void, BlobSetTierHeaders> callback) {
-        storageBlobServiceClient.setBlobTier(containerName,
-            blobName,
-            tier,
-            callback);
+    public void setBlobAccessTier(@NonNull String containerName,
+                                  @NonNull String blobName,
+                                  @Nullable AccessTier tier,
```

Comment:
I think tier is required and should be NonNull

Reply:
Right. Thanks for catching that